### PR TITLE
Documentation with instructions for fork and code

### PR DESCRIPTION
**Pull request by jananivMS**
Closes issue #290 

**What this PR does / why we need it**:
This PR adds documentation with instructions for a developer to fork this repo and make changes to easily build their app.

**Special notes for your reviewer**:
- If there are steps are too verbose or not needed please feel free to comment.
- If the instructions aren't clear, please comment

**If applicable**:
- [X] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

File: docs/fork-and-code-using-helium.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Instructions to fork and code using this repo
+
+## Overview
+If you are starting to write a Node.JS API server, the below steps will help you fork this repo and easily get started on your app while incorporating best practices. 
+The instructions below detail the changes you would need to make for your app after forking the code.
+
+## Before you start
+This repo uses Cosmos DB for the database component, Azure Application Insights for telemetry and Bunyan Logger for tracing.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
This repo uses 
- [Cosmos DB](https://azure.microsoft.com/en-us/services/cosmos-db/) for the database component, 
- [Azure Application Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview) for telemetry 
- [Bunyan Logger](https://github.com/trentm/node-bunyan) for tracing

```

File: docs/fork-and-code-using-helium.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Overview
+If you are starting to write a Node.JS API server, the below steps will help you fork this repo and easily get started on your app while incorporating best practices. 
+The instructions below detail the changes you would need to make for your app after forking the code.
+
+## Before you start
+This repo uses Cosmos DB for the database component, Azure Application Insights for telemetry and Bunyan Logger for tracing. 
+You can continue to use these or replace them with a provider of your choice.
+
+## Prerequisites
+The instructions here do not cover the steps to follow to import data into your database. The assumption is that you will have decided the schema for the data and added records to the database.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]: We could link them to the 4co repo with the example data we are using. There are instructions there.

https://github.com/4-co/imdb

File: docs/fork-and-code-using-helium.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+This repo uses Cosmos DB for the database component, Azure Application Insights for telemetry and Bunyan Logger for tracing. 
+You can continue to use these or replace them with a provider of your choice.
+
+## Prerequisites
+The instructions here do not cover the steps to follow to import data into your database. The assumption is that you will have decided the schema for the data and added records to the database. 
+
+## Step-by-Step instructions
+1. Fork this repo by navigating to https://github.com/microsoft/helium and clicking on the "Fork" button on the top right.
+
+![image](imgs/forkrepo.png)
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]: This image's name is actually `forkrepo.PNG` and this renderer is case sensitive.

File: docs/fork-and-code-using-helium.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Once forked, clone the forked repo on to your computer using the 'git clone' command.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
Once forked, clone the forked repo onto your computer using the `git clone` command.
```

File: docs/fork-and-code-using-helium.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+2. Define models for the entities. Based on the schema of the data in your database design the models for the different entities you will be operating on. For example, in our sample, we have 'Movie', 'Actor', 'Genre' as the entities.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
2. Model the entities your API will expose by creating classes in `/src/app/models`. For examples, see the Movie, Actor, and Genre models created for the reference app [here](/src/app/models).
```

File: docs/fork-and-code-using-helium.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+3. Navigate to the src/app/models folder and add a separate .ts file for each entity in your app. Look for the **[UPDATE_AFTER_FORK]** comment and update based on your entity.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]: Not a huge fan of having these extra comments in the code. Can we get away with just describing the places that need updating?

File: docs/fork-and-code-using-helium.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+4. Define what operations you would like to perform on the entities. For example, in our case, we perfom "GET all", "GET by Id" and "POST" operations on the "Actor" entity. 
+
+5. Navigate to the src/app/controllers folder and add a separate .ts for each entity which will define all the operations for that entity. You can copy over the code from one of the files under the controllers folder to this file and update it.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
5. Navigate to the `src/app/controllers` folder and add a separate `.ts` file for each entity which will define all the operations for that entity. You can copy over the code from one of the files under the controllers folder to this file and update it.
```

File: docs/fork-and-code-using-helium.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+6. Look for the **[UPDATE_AFTER_FORK]** comment and update the decorators to reflect the operation and the path based on your API.
+
+7. Leave the system.ts file as we recommend implementing a health check endpoint in your app. You can make changes to the actual implementation of the function based on the **[UPDATE_AFTER_FORK]** comment in the file.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
7. Leave the `system.ts` file as we recommend implementing a health check endpoint in your app. You can make changes to the actual implementation of the function based on the **[UPDATE_AFTER_FORK]** comment in the file. 
```

File: docs/fork-and-code-using-helium.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+4. Define what operations you would like to perform on the entities. For example, in our case, we perfom "GET all", "GET by Id" and "POST" operations on the "Actor" entity. 
+
+5. Navigate to the src/app/controllers folder and add a separate .ts for each entity which will define all the operations for that entity. You can copy over the code from one of the files under the controllers folder to this file and update it.
+
+6. Look for the **[UPDATE_AFTER_FORK]** comment and update the decorators to reflect the operation and the path based on your API.
+
+7. Leave the system.ts file as we recommend implementing a health check endpoint in your app. You can make changes to the actual implementation of the function based on the **[UPDATE_AFTER_FORK]** comment in the file. 
+
+8. Navigate to the src/app/tests folder and add one .ts file for each controller. Write the integration tests modelled on what is in our sample.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
8. Navigate to the `src/app/tests` folder and add one `.ts` file for each controller. Write the integration tests modelled on what is in our sample.
```

File: docs/fork-and-code-using-helium.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+9. Remove the files from our sample that don't apply to your app from src/app/models, src/app/controllers and src/app/tests folders.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
9. Remove the files from our sample that don't apply to your app from `src/app/models`, `src/app/controllers` and `src/app/tests` folders.
```

File: docs/fork-and-code-using-helium.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+10. The next step is to update the src/app/server.ts file. Look for the **[UPDATE_AFTER_FORK]** comment and update the code to bind the controllers that you have implemented.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
10. The next step is to update the `src/app/server.ts` file. Look for the **[UPDATE_AFTER_FORK]** comment and update the code to bind the controllers that you have implemented. 
```

File: docs/fork-and-code-using-helium.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+7. Leave the system.ts file as we recommend implementing a health check endpoint in your app. You can make changes to the actual implementation of the function based on the **[UPDATE_AFTER_FORK]** comment in the file. 
+
+8. Navigate to the src/app/tests folder and add one .ts file for each controller. Write the integration tests modelled on what is in our sample.
+
+9. Remove the files from our sample that don't apply to your app from src/app/models, src/app/controllers and src/app/tests folders.
+
+10. The next step is to update the src/app/server.ts file. Look for the **[UPDATE_AFTER_FORK]** comment and update the code to bind the controllers that you have implemented. 
+If you don't use the default providers for database, telemetry and logging, you will need to bind those controllers appropriately too. 
+
+11. After this you can follow the steps in (deploying-helium.md) to build and test the app.
```
Comment (frodopwns) [inline docs/fork-and-code-using-helium.md]:
```suggestion
11. After this you can follow the steps in [deploying-helium](deploying-helium.md) to build and test the app.
```

File: docs/fork-and-code-using-helium.md (hunk 1 above)
Comment (jananivMS) [reply docs/fork-and-code-using-helium.md]: Looks good, updated

File: docs/fork-and-code-using-helium.md (hunk 6 above)
Comment (jananivMS) [reply docs/fork-and-code-using-helium.md]: This is something I am not sure of if we could do a good job of explaining the place. Would like to discuss before updating.

**Pull request merged**